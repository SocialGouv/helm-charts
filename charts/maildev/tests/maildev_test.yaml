suite: test maildev
#templates:
tests:
  - it: should render
    asserts:
      - template: deployment.yaml
        isKind:
          of: Deployment
      - template: deployment.yaml
        hasDocuments:
          count: 1
  - it: default manifest
    asserts:
      - template: deployment.yaml
        matchSnapshot:
          path: spec
      - template: pvc.yaml
        hasDocuments:
          count: 0
      - template: cron.yaml
        hasDocuments:
          count: 0
  - it: custom resources
    values:
      - values-resources.yaml
    asserts:
      - template: deployment.yaml
        matchSnapshot:
          path: spec
      - template: pvc.yaml
        hasDocuments:
          count: 0
  - it: persistence enabled
    values:
      - values-persistence.yaml
    asserts:
      - template: deployment.yaml
        matchSnapshot:
          path: spec
      - template: pvc.yaml
        matchSnapshot:
          path: spec
      - template: cron.yaml
        hasDocuments:
          count: 0
  - it: persistence and existingClaim enabled
    values:
      - values-persistence-existingClaim.yaml
    asserts:
      - template: deployment.yaml
        matchSnapshot:
          path: spec
      - template: pvc.yaml
        hasDocuments:
          count: 0
      - template: cron.yaml
        hasDocuments:
          count: 0
  - it: cron enabled
    values:
      - values-persistence-cron.yaml
    asserts:
      - template: deployment.yaml
        matchSnapshot:
          path: spec
      - template: pvc.yaml
        matchSnapshot:
          path: spec
      - template: cron.yaml
        matchSnapshot:
          path: spec
