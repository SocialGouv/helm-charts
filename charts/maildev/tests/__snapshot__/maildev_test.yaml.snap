cron enabled:
  1: |
    replicas: 1
    selector:
      matchLabels:
        component: maildev
    template:
      metadata:
        labels:
          component: maildev
      spec:
        containers:
          - args:
              - --mail-directory
              - /home/node/maildev-data
            image: maildev/maildev:2.1.0
            imagePullPolicy: IfNotPresent
            livenessProbe:
              httpGet:
                path: /healthz
                port: http
            name: maildev
            ports:
              - containerPort: 1080
                name: http
                protocol: TCP
              - containerPort: 1025
                name: smtp
                protocol: TCP
            readinessProbe:
              httpGet:
                path: /healthz
                port: http
            resources:
              limits:
                cpu: null
                memory: null
              requests:
                cpu: null
                memory: null
            volumeMounts:
              - mountPath: /home/node/maildev-data
                name: data
        volumes:
          - name: data
            persistentVolumeClaim:
              claimName: maildev
  2: |
    accessModes:
      - ReadWriteMany
    resources:
      requests:
        storage: 1024M
  3: |
    jobTemplate:
      spec:
        template:
          spec:
            containers:
              - args:
                  - find
                  - /home/node/maildev-data
                  - -type
                  - f
                  - -mtime
                  - "+15"
                  - -exec
                  - rm
                  - -f
                  - '{}'
                  - ;
                image: busybox:latest
                imagePullPolicy: IfNotPresent
                name: delete-mail
                volumeMounts:
                  - mountPath: /home/node/maildev-data
                    name: data
              - args:
                  - wget
                  - maildev:1080/reloadMailsFromDirectory
                image: busybox:latest
                imagePullPolicy: IfNotPresent
                name: reload-mail
            restartPolicy: OnFailure
            volumes:
              - name: data
                persistentVolumeClaim:
                  claimName: maildev
    schedule: 0 1 * * *
    startingDeadlineSeconds: 100
custom resources:
  1: |
    replicas: 1
    selector:
      matchLabels:
        component: maildev
    template:
      metadata:
        labels:
          component: maildev
      spec:
        containers:
          - args:
              - --mail-directory
              - /home/node/maildev-data
            image: maildev/maildev:2.1.0
            imagePullPolicy: IfNotPresent
            livenessProbe:
              httpGet:
                path: /healthz
                port: http
            name: maildev
            ports:
              - containerPort: 1080
                name: http
                protocol: TCP
              - containerPort: 1025
                name: smtp
                protocol: TCP
            readinessProbe:
              httpGet:
                path: /healthz
                port: http
            resources:
              limits:
                cpu: 42m
                memory: 100m
              requests:
                cpu: 12m
                memory: 10m
            volumeMounts:
              - mountPath: /home/node/maildev-data
                name: data
        volumes:
          - emptyDir: {}
            name: data
default manifest:
  1: |
    replicas: 1
    selector:
      matchLabels:
        component: maildev
    template:
      metadata:
        labels:
          component: maildev
      spec:
        containers:
          - args:
              - --mail-directory
              - /home/node/maildev-data
            image: maildev/maildev:2.1.0
            imagePullPolicy: IfNotPresent
            livenessProbe:
              httpGet:
                path: /healthz
                port: http
            name: maildev
            ports:
              - containerPort: 1080
                name: http
                protocol: TCP
              - containerPort: 1025
                name: smtp
                protocol: TCP
            readinessProbe:
              httpGet:
                path: /healthz
                port: http
            resources:
              limits:
                cpu: null
                memory: null
              requests:
                cpu: null
                memory: null
            volumeMounts:
              - mountPath: /home/node/maildev-data
                name: data
        volumes:
          - emptyDir: {}
            name: data
persistence and existingClaim enabled:
  1: |
    replicas: 1
    selector:
      matchLabels:
        component: maildev
    template:
      metadata:
        labels:
          component: maildev
      spec:
        containers:
          - args:
              - --mail-directory
              - /home/node/maildev-data
            image: maildev/maildev:2.1.0
            imagePullPolicy: IfNotPresent
            livenessProbe:
              httpGet:
                path: /healthz
                port: http
            name: maildev
            ports:
              - containerPort: 1080
                name: http
                protocol: TCP
              - containerPort: 1025
                name: smtp
                protocol: TCP
            readinessProbe:
              httpGet:
                path: /healthz
                port: http
            resources:
              limits:
                cpu: null
                memory: null
              requests:
                cpu: null
                memory: null
            volumeMounts:
              - mountPath: /home/node/maildev-data
                name: data
        volumes:
          - name: data
            persistentVolumeClaim:
              claimName: maildev
persistence enabled:
  1: |
    replicas: 1
    selector:
      matchLabels:
        component: maildev
    template:
      metadata:
        labels:
          component: maildev
      spec:
        containers:
          - args:
              - --mail-directory
              - /home/node/maildev-data
            image: maildev/maildev:2.1.0
            imagePullPolicy: IfNotPresent
            livenessProbe:
              httpGet:
                path: /healthz
                port: http
            name: maildev
            ports:
              - containerPort: 1080
                name: http
                protocol: TCP
              - containerPort: 1025
                name: smtp
                protocol: TCP
            readinessProbe:
              httpGet:
                path: /healthz
                port: http
            resources:
              limits:
                cpu: null
                memory: null
              requests:
                cpu: null
                memory: null
            volumeMounts:
              - mountPath: /home/node/maildev-data
                name: data
        volumes:
          - name: data
            persistentVolumeClaim:
              claimName: maildev
  2: |
    accessModes:
      - ReadWriteMany
    resources:
      requests:
        storage: 1024M
